http.c

```diff
@@ -415,18 +464,17 @@ int http_get(struct http_get *hg, struct keyval *headers, int timeout_ms)
 	return rc;
 }
 
-char *http_read_body(int fd, size_t *size, int timeout_ms)
+char *http_read_body(struct connection *conn, size_t *size, int timeout_ms)
 {
 	GBUF(buf);
-
-	if (read_timeout(fd, timeout_ms))
+	if (read_timeout(get_sockfd(conn), timeout_ms))
 		return NULL;
 	while (1) {
 		int count = 1023;
```

Comment:
Odd number.

Reply:
This number is not from me and has been here for quite some time.
Isn't it a way to keep the buffer null-terminated and aligned?

I don't think there's much benefit to changing this value without further information.